frame/oracle/src/lib.rs

```diff
@@ -457,23 +461,27 @@ pub mod pallet {
 		pub fn add_asset_and_info(
 			origin: OriginFor<T>,
 			asset_id: T::AssetId,
-			threshold: Percent,
-			min_answers: u32,
-			max_answers: u32,
-			block_interval: T::BlockNumber,
+			valid_threshold: Validated<Percent, ValidThreshhold>,
+			valid_min_answers: Validated<u32, ValidMinAnswers>,
+			valid_max_answers: Validated<u32, ValidMaxAnswer<T::MaxAnswerBound>>,
+			valid_block_interval: Validated<T::BlockNumber, ValidBlockInterval<T::StalePrice>>,
 			reward: BalanceOf<T>,
 			slash: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
 			T::AddOracle::ensure_origin(origin)?;
-			ensure!(min_answers > 0, Error::<T>::InvalidMinAnswers);
+
+			let threshold = valid_threshold.value();
+			let min_answers = valid_min_answers.value();
+			let max_answers = valid_max_answers.value();
+			let block_interval = valid_block_interval.value();
```

Comment:
Can we make the Deref impl public to avoid the need to remap names?

Reply:
deref would eventually lead to clones if we use move:
```
let (x,y,z) = (z.into(), y.into(), z.into());
```
or as in lending - if there are so much parameters (and new may be added for v2) - create struct.

Reply:
they are thinking about DerefMove for several years:)

Reply:
oh, in this case it is all copy values. will work.

Reply:
fixed in 43da2998d2db1e50149a6882807743a7684de31e